### PR TITLE
docs: Cloud Run / GCP Serverless

### PR DESCRIPTION
**Pull request by travisgroth**
## Summary
Add documentation related to Cloud Run and GCP Serverless headers.

- core docs
- example
- recipe

## Related issues
#1080 

**Checklist**:
- [x] add related issues
- [x] updated docs
- [x] ready for review

### CODE REVIEW

File: examples/cloudrun/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Pomerium on Cloud Run
+
+Run this demo with gcloud command line configured for your project.  The commands assume 
+all resources (Cloud Run, Cloud DNS, and Secret Manager) are in a single project.  
+
+We recommend dedicated project that is easy to clean up.
```
Comment (desimone) [inline examples/cloudrun/README.md:6]: > We recommend dedicated

> We recommend a dedicated

File: docs/recipes/cloud-run.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  This guide covers how to deploy Pomerium to Cloud Run and use it to protect other endpoints via Authorization Headers.
+---
+
+# Securing Cloud Run endpoints
+
+This recipe's sources can be found [on github](https://github.com/pomerium/pomerium/tree/master/examples/cloudrun)
+
+## Background
+
+Services on Cloud Run and other Google Cloud serverless products can be restricted to only permit access with a properly signed [bearer token](https://cloud.google.com/run/docs/authenticating/service-to-service).  This allows requests from other services running in GCP or elsewhere to be securely authorized despite the endpoints being public.
```
Comment (desimone) [inline docs/recipes/cloud-run.md:17]: maybe link to cloud run for the first time we mention it

File: docs/recipes/cloud-run.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+These bearer tokens are not easily set in a browser session and must be refreshed on a regular basis, preventing them from being useful for end user authorization.  Pomerium, however, can generate compatible tokens on behalf of end users and proxy the request to these services.  
+
+## How it works
+
+- Add an IAM policy delegating `roles/run.invoker` permissions to a service account
+- Run Pomerium with access to a key for the corresponding service account
+- Publish DNS records for each protected application pointing to Pomerium
+- Configure Pomerium with appropriate policy and `enable_google_cloud_serverless_authentication`
+
+The protected application delegates trust to a GCP service account which Pomerium runs as, and Pomerium performs user based authorization on a per route basis.  This turns Pomerium into a bridge between a user-centric and service-centric authorization models.
```
Comment (desimone) [inline docs/recipes/cloud-run.md:28]: Well put.

File: docs/recipes/cloud-run.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+## Pre-requisites
+
+This guide assumes you have Editor access to a Google Cloud project which can be used for isolated testing, and a DNS zone which you are also able to control.  DNS does not need to be inside Google Cloud for the example to work.
+
+## Set Up
+
+To deploy Pomerium to Cloud Run securely and easily, a special image is maintained at `gcr.io/pomerium-io/pomerium-[version]-cloudrun`.  It allows sourcing configuration from GCP Secrets Manager, and sets some defaults for Cloud Run to keep configuration minimal.  We will be leveraging it in this example to store IdP credentials.
```
Comment (desimone) [inline docs/recipes/cloud-run.md:36]: Maybe link to the docker definition?

Comment (travisgroth) [reply docs/recipes/cloud-run.md]: Added a bit more context here with both links and some more detail.

**Comment by travisgroth** [comment]
>And maybe maybe we can also add a summary for the landing page for recipes on

👍

**Pull request merged**